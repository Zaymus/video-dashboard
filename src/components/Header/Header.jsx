import styled from "styled-components";
import SearchBar from "../SearchBar/";
import { NavLink } from "react-router";

const HeaderContainer = styled.div`
  width: 100%;
  padding: 0 15px;
  height: var(--header-height);
  background: var(--background-dark);
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid var(--primary);
  position: fixed;
  top: 0;
  left: 0;
  box-sizing: border-box;

  a {
    color: var(--primary);
    text-decoration: none;
    transition: text-shadow 150ms linear;
    
    &:active {
      color: var(--primary);
    }

    &:visitied {
      color: var(--primary);
    }

    &:hover {
      text-shadow: 0px 0px 10px var(--primary);
    }
  }
`;

const Header = () => {
  return (
    <HeaderContainer>
      <NavLink to="/">Video Dashboard</NavLink>
      <SearchBar />
    </HeaderContainer>
  );
}

export default Header;