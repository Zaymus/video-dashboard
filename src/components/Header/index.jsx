import { lazy } from 'react';

const Header = lazy(() => import('./Header.jsx'));

export default Header;