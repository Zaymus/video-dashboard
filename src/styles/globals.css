:root {
  --primary: #1E90FF;
  --secondary: #FF6B6B;
  --background-dark: #121212;
  --surface-dark: #1F1F1F;
  --text-primary-dark: #FFFFFF;
  --text-secondary-dark: #B3B3B3;
  --border-dark: #2C2C2C;
  --success: #2ECC71;
  --error: #E74C3C;
  --warning: #F1C40F;
  --background-light: #F9F9F9;
  --surface-light: #FFFFFF;
  --text-primary-light: #1A1A1A;
  --text-secondary-light: #4F4F4F;
  --border-light: #E0E0E0;

  --videos-gutter-margin: 30px;
  --videos-per-row: 5;
  --videos-usable-width: calc(100% - var(--videos-gutter-margin) * 2);
}

body {
  background-color: var(--background-dark);
  margin: 0;
  color: var(--text-primary-dark);
  font-size: 16pt;
}

#root {
  min-height: 100vh;
}