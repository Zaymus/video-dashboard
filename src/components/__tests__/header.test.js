import { describe, test, expect } from '@jest/globals';
import { render } from '@testing-library/react';

import Header from '../Header/Header';

describe('Header component test', () => {
  test('matches snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  })
});