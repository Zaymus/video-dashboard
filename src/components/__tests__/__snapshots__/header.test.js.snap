// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header component test matches snapshot 1`] = `
<div>
  <div
    class="sc-ggWZvA QVSmT"
  >
    <div
      class="sc-bRKDuR ecbMmo"
    >
      <input
        class="sc-hvigdm iMebNp"
        name="search"
        placeholder="Search"
        spellcheck="true"
        type="search"
      />
      <button
        class="sc-fhHczv jvyIXk"
      >
        <i
          class="fa-solid fa-magnifying-glass"
        />
      </button>
    </div>
  </div>
</div>
`;
